### PR TITLE
[GPU] print datashape of input for benchmark_app

### PR DESCRIPTION
**Pull request by sungeunk**
-added option: OV_GPU_PrintLayersInput

### Tickets:
 - 123974

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/runtime/debug_configuration.hpp [hunk 1]
```diff
@@ -113,6 +113,7 @@ class debug_configuration {
     int dump_layers_dst_only;                                   // Dump only output of layers
     int dump_layers_result;                                     // Dump result layers
     int dump_layers_input;                                      // Dump input layers
+    int print_layers_input;                                     // Print the input data_shape for benchmark_app. This option is related to dump_iteration.
```
Comment (isanghao) [inline src/plugins/intel_gpu/include/intel_gpu/runtime/debug_configuration.hpp:116]: * please move it to different place
* What about "print_input_data_shapes" instead

File: src/plugins/intel_gpu/src/graph/network.cpp [hunk 2]
```diff
@@ -1415,6 +1415,28 @@ void network::execute_impl(const std::vector<event::ptr>& events) {
...
+            auto name = inst->id();
+            auto pos = name.find(':');
+            auto type = name.substr(0, pos);
+            name.erase(0, pos + 1);
+            if (inst->is_input() && type == "parameter") {
+                add_string(name + inst->get_output_layout().get_partial_shape().to_string());
+            }
+        }
+
+        GPU_DEBUG_COUT << "[Bprogram:" << std::setw(2) << ((get_program() != nullptr) ? get_program()->get_id() : 0)
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/network.cpp:1435]: why it starts with `[B`?

File: src/plugins/intel_gpu/include/intel_gpu/runtime/debug_configuration.hpp (hunk 1 above)
Comment (sungeunk) [reply src/plugins/intel_gpu/include/intel_gpu/runtime/debug_configuration.hpp]: updated PR

File: src/plugins/intel_gpu/src/graph/network.cpp (hunk 2 above)
Comment (sungeunk) [reply src/plugins/intel_gpu/src/graph/network.cpp]: updated PR

**Review by isanghao — APPROVED** [review]

**Pull request merged**